enhancements/network/enable-flows-collection.md

```diff
@@ -0,0 +1,352 @@
+                       +-----------+-+
+
+                                
+    +------------------------------------------------+
+    |  NAME       STATUS      ENDPOINT               |
+    +------------------------------------------------+
+    |  node-a     emitting    http://10.1.2.3:30001  |
+    |  node-b     emitting    http://10.4.5.6:30001  | 
+    |  node-c     emitting    http://10.7.8.9:30001  |
+    +------------------------------------------------+
+```
+
+#### Tuning flows' emission parameters
+
+Some setups might lead to an overcongestion of the cluster resources such as the network or the
+flows' ingestion/processing/storage pipeline (e.g. because the size and frequency of reported
+flows is too high).
+
+To minimize it, the user should be able to tune up the flows' cache mechanism in order to control
+the number of records sent to collector.
+
+According to the [OpenVSwitch configuration options](https://www.openvswitch.org/support/dist-docs/ovs-vswitchd.conf.db.5.html),
+there are multiple configuration options that could be exposed to the user:
+
+* `cache_active_timeout`: optional integer, in range 0 to 4,200
+  The maximum period in seconds for which an IPFIX flow record  is
+  cached  and  aggregated  before  being  sent.  If not specified,
+  defaults to 0. If 0, caching is disabled.
+
+* `cache_max_flows`: optional integer, in range 0 to 4,294,967,295
+  The maximum number of IPFIX flow records that can be cached at a
+  time.  If  not  specified,  defaults to 0. If 0, caching is disabled.
+
+* `sampling`: optional integer, in range 1 to 4,294,967,295
+  The  rate  at  which  packets should be sampled and sent to each
+  target collector. If not specified, defaults to 400, which means
+  one  out of 400 packets, on average, will be sent to each target
+  collector.
+
+Other arguments might be considered, such as `enable-input-sampling`, `enable-output-sampling`,
+or `enable-tunnel-sampling`.
+
+The configuration should be also exposed in the Console UI, along with the current
+emission status (see previous subsection).
+
+### Implementation Details/Notes/Constraints [optional]
+
+In order to ensure an optimal usage of the network, we should restrict flow's
+emission-collection traffic to the local node.
+
+According to the [architectural description of the flows' emitter/collectors pipeline](
+https://docs.google.com/document/d/1kzNfTLXhMolu8VTcH0m9hmvFluN9yKMy7TSNIXJVTrA/edit#heading=h.kulf936ct32p),
+the collector will be deployed as a DaemonSet, to ensure that there will be an
+instance of the collector for each node.
+
+Exposing the collector endpoint as a Kubernetes ClusterIP Service would cause the traffic
+to be distributed across all the nodes (unless you enable the `ServiceInternalTrafficPolicy`
+cluster-wide feature gate, which at this moment is in alpha status, only available
+for the latest Kubernetes 1.21 version).
+
+To limit the flows' traffic to internal nodes, each collector should expose a node
```

Comment:
My understanding here is that we could use directly the host IP, not the service IP, to avoid LB. The Service here is not really mandatory, it's only used to declare the port to be used. Probably we shouldn't use a Service at all and declare the port in the exporter config, rather than having a diverted use of a Service.

Reply:
Joel is right. I'm rewriting it to avoid any misunderstanding.